File: Web-Frontend/src/api/projectApi.js
import API from "./axios";


export const getProjects = (params) => API.get("/projects", { params });
export const getProjectById = (id) => API.get(`/projects/${id}`);
export const getProjectStats = () => API.get("/projects/stats");

export const createProject = (formData) =>
  API.post("/projects", formData, {
    headers: { "Content-Type": "multipart/form-data" },
  });

export const updateProject = (id, formData) =>
  API.put(`/projects/${id}`, formData, {
    headers: { "Content-Type": "multipart/form-data" },
  });

export const cancelProject = (id) => API.patch(`/projects/${id}/cancel`);

export const approveProject = (id) => API.patch(`/projects/${id}/approve`);

export const rejectProject = (id, comment) =>
  API.patch(`/projects/${id}/reject`, { comment });

File: Mobile-App/mint_mobile/lib/api/project_api.dart
import 'package:dio/dio.dart';
import 'dio_client.dart';

class ProjectApi {
  static Future<Response> getProjects() =>
      DioClient.instance.get('/projects');

  static Future<Response> getProjectStats() =>
      DioClient.instance.get('/projects/stats');

  static Future<Response> createProject(FormData formData) =>
      DioClient.instance.post(
        '/projects',
        data: formData,
        options: Options(contentType: 'multipart/form-data'),
      );

  static Future<Response> updateProject(String id, FormData formData) =>
      DioClient.instance.put(
        '/projects/$id',
        data: formData,
        options: Options(contentType: 'multipart/form-data'),
      );

  static Future<Response> cancelProject(String id) =>
      DioClient.instance.patch('/projects/$id/cancel');

  static Future<Response> approveProject(String id) =>
      DioClient.instance.patch('/projects/$id/approve');

  static Future<Response> rejectProject(String id, String comment) =>
      DioClient.instance.patch('/projects/$id/reject', data: {'comment': comment});
}
